web-common/src/features/embeds/embed-theme-utils.ts

```diff
@@ -0,0 +1,25 @@
+import { EmbedStore } from "./embed-store";
+import { getEmbedThemeStoreInstance } from "./embed-theme-store";
+import { get } from "svelte/store";
+
+/**
+ * Resolves the theme name for embeds with priority:
+ * 1. API value (including explicit null)
+ * 2. Embed theme store value
+ * 3. Initial embed URL theme
+ */
+export function resolveEmbedTheme(
+  embedThemeValue?: string | null,
+): string | null {
+  // 1. Explicit API-provided value (including null to clear)
+  if (embedThemeValue !== undefined) return embedThemeValue;
+
+  // 2. Value from the scoped embed theme store, if non-null
+  const embedThemeStore = getEmbedThemeStoreInstance();
```

Comment:
Since this method is always called with `embedThemeStore` already do we still need this?